maven-core/src/main/java/org/apache/maven/ReactorReader.java

```diff
@@ -64,28 +63,21 @@
 class ReactorReader implements MavenWorkspaceReader {
     public static final String HINT = "reactor";
 
-    private static final Collection<String> COMPILE_PHASE_TYPES =
-            Arrays.asList("jar", "ejb-client", "war", "rar", "ejb3", "par", "sar", "wsr", "har", "app-client");
-
     private static final Logger LOGGER = LoggerFactory.getLogger(ReactorReader.class);
 
     private final MavenSession session;
     private final Map<String, MavenProject> projectsByGAV;
     private final Map<String, List<MavenProject>> projectsByGA;
     private final WorkspaceRepository repository;
 
-    private Function<MavenProject, String> projectIntoKey =
-            s -> ArtifactUtils.key(s.getGroupId(), s.getArtifactId(), s.getVersion());
-
-    private Function<MavenProject, String> projectIntoVersionlessKey =
-            s -> ArtifactUtils.versionlessKey(s.getGroupId(), s.getArtifactId());
-
     @Inject
     ReactorReader(MavenSession session) {
         this.session = session;
-        this.projectsByGAV = session.getAllProjects().stream().collect(toMap(projectIntoKey, identity()));
+        this.projectsByGAV = session.getAllProjects().stream()
```

Comment:
I don't see how this is necessary for this issue, but I think it is equivalent from a technical point of view.

Reply:
It's not.  I'll revert this hunk to minimize the changes.